shell/platform/android/io/flutter/embedding/android/FlutterView.java

```diff
@@ -1407,10 +1426,34 @@ public void removeFlutterEngineAttachmentListener(
             ? SettingsChannel.PlatformBrightness.dark
             : SettingsChannel.PlatformBrightness.light;
 
+    boolean isNativeSpellCheckServiceDefined = false;
+
+    if (textServicesManager != null) {
+      if (Build.VERSION.SDK_INT >= 31) {
+        List<SpellCheckerInfo> enabledSpellCheckerInfos =
+            textServicesManager.getEnabledSpellCheckerInfos();
+        boolean gboardSpellCheckerEnabled =
+            enabledSpellCheckerInfos.stream()
+                .anyMatch(
+                    spellCheckerInfo ->
+                        spellCheckerInfo
+                            .getPackageName()
+                            .equals("com.google.android.inputmethod.latin"));
```

Comment:
is this detecting if the spell checker is for latin-only languages?  what part makes this Gboard specific? What about for Hindi, Japanese, etc...?  I think it's ok to start with latin only, but we would probably need to file a bug after confirming the behavior for non-latin languages.

Reply:
As far as I can tell, this package covers all of the languages that the Gboard spell checker supports, which is also the only spell checker I've actually tested. This check was an accident on my part -- I confused the spell checker(s) that Android supports with this spell checker.

I can remove this check. Alternatively, I can leave it and file a bug, as you suggested, to verify the behavior of other spell checkers.

Reply:
bug filed works